### DISCUSSION TITLE
How to change color based on the category?

### DISCUSSION THREAD

**Discussion by sasori13L** [saade/filament-fullcalendar#172] [Q&A]
![image](https://github.com/saade/filament-fullcalendar/assets/16425810/39422b09-85d8-44c3-a5ef-73fa44496d03)

How can I change the color of each entry? the FullCalendar.io site has instructions on how to do it but I don't know how to apply it using this plugin.

**Comment by olivier-nolf** [comment]
Hi,

You can use the eventClassNames to add a custom className to your event.

public function eventClassNames(): string
    {
  
      $categories= [
            'cat1' => 'class1',
            'cat2' => 'class2',
            'cat3' => 'class3',
            'cat4' => 'class4',
        ];
    
        $jsonColorOptions = json_encode($categories);
        return <<<JS
            function({ event, timeText, isStart, isEnd, isMirror, isPast, isFuture, isToday, el, view }){
                var categories= $jsonColorOptions;
                var className = categories[event.extendedProps.type];
    
                return [
                   className;
                ]
            }
        JS;
    } 
    
    And of course creating a custom theme in filament or something similar.

**Comment by saade** [comment]
Look at EventData, there's a background and text color configuration
